tests/integration/components/file-menu-test.js

```diff
@@ -0,0 +1,132 @@
+import Ember from "ember";
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+let file, model;
+
+let addFileCalled = false;
+let renameFileCalled = false;
+let removeFileCalled = false;
+let saveGistCalled = false;
+let forkCalled = false;
+let deleteGistCalled = false;
+let signInViaGithubCalled = false;
+
+let fileType = null;
+let renamedFile = null;
+let removedFile = null;
+let gistToSave = null;
+let gistToFork = null;
+let gistToDelete = null;
+
+moduleForComponent('file-menu', 'Integration | Component | file menu', {
+  integration: true,
+  beforeEach() {
+
+    // Set any properties with this.set('myProperty', 'value');
+    file = Ember.Object.create({
+      filePath: "some.path.js"
+    });
+
+    model = Ember.Object.create({
+      id: '74bae9a34142370ff5a3',
+      files: [file],
+      history: [],
+      isNew: false
+    });
+
+    this.set('model', model);
+
+    this.set('session', Ember.Object.create({
+      isAuthenticated: true
+    }));
+
+    this.set('activeEditorCol', 1);
+
+    this.set('activeFile', file);
+
+    // Handle any actions with this.on('myAction', function(val) { ... });
+    this.on('addFile', (type) => { addFileCalled = true; fileType = type; });
```

Comment:
Can you tweak these to save to `this`?  Something like:

```javascript
this.on('addFile', (type) => { this.addFileCalled = true; this.fileType = type; });
```

Then you can assert in the tests below like:

```javascript
assert.ok(this.addFileCalled, 'addFile was called');
assert.equal(this.fileType, 'template', 'addFile was called with type parameter');
```

---

Using the above, will help us ensure that there is no leakage between tests.  Say we add a test in the future that also triggers an action, the boolean `xyzCalled` may have already been set causing the tests to be not quite right.